### PR TITLE
Add kirigami drawer

### PR DESCRIPTION
**Pull request by JBBgameich**
This adds a drawer for switching between future chat and contacts and similar.
kaidan
The logo at the top left corner is from kopete and will be replaced by our new one when we have decided on one. The banner was made by github.com/mbblp. (How will we handle credits and media licenses?)
There are also options on the drawer which doesn't have a function now, but some already work, like going back to login in the settings action. This is work in progress, so there will follow more.
Things like not opening the drawer at start when the width space is to small are't implemented yet.
The drawer is also a global drawer, we could discuss about hidding it while login.

### CODE REVIEW

**Comment by LNJ2** [comment]
1. The drawer should definitely not show up on the log in page.
2. You should need to go into the drawer to go to the contacts/chat page

3. I'll do the licensing part, if that's ok. (@mbblp CC BY-SA 4.0?)

File: kaidan.qrc [hunk 1]
```diff
@@ -3,5 +3,6 @@
 		<file alias="main.qml">src/qml/main.qml</file>
 		<file alias="RosterPage.qml">src/qml/RosterPage.qml</file>
 		<file alias="LoginPage.qml">src/qml/LoginPage.qml</file>
+		<file alias="banner.png">data/images/banner.png</file>
```
Comment (LNJ2) [inline kaidan.qrc]: I think we should split media and code, better name the alias "data/banner.png".

File: src/qml/main.qml [hunk 2]
```diff
@@ -5,6 +5,52 @@ import harbour.kaidan 1.0
 Kirigami.ApplicationWindow {
 	id: root
 	visible: true
+	
+	globalDrawer: Kirigami.GlobalDrawer {
```
Comment (LNJ2) [inline src/qml/main.qml]: We shouldn't use a global drawer...

File: src/qml/main.qml [hunk 3]
```diff
@@ -5,6 +5,52 @@ import harbour.kaidan 1.0
...
+			Kirigami.Action {
+				text: "login"
+				iconName: "system-switch-user"
+				onTriggered: applicationWindow().pageStack.replace(loginPageComponent)
+			}
+		},
+		Kirigami.Action {
+			text: "Contacts"
+			iconName: "x-office-contact.svg"
+            onTriggered: applicationWindow().pageStack.replace(rosterPageComponent)
```
Comment (LNJ2) [inline src/qml/main.qml]: tabs please

File: src/qml/main.qml [hunk 4]
```diff
@@ -5,6 +5,52 @@ import harbour.kaidan 1.0
...
 	id: root
 	visible: true
+	
+	globalDrawer: Kirigami.GlobalDrawer {
+		title: "Kaidan"
+		titleIcon: "kopete"
+		bannerImageSource: "qrc:/banner.png"
+
+		actions: [
+		Kirigami.Action {
```
Comment (LNJ2) [inline src/qml/main.qml]: Here is one tab missing. (till L. 51)

File: src/qml/main.qml [hunk 5]
```diff
@@ -5,6 +5,52 @@ import harbour.kaidan 1.0
...
+			text: "My Profile"
+			iconName: "view-list-icons"
+		},
+		Kirigami.Action {
+			text: "Settings"
+			iconName: "menu_new"
+			Kirigami.Action {
+				text: "login"
+				iconName: "system-switch-user"
+				onTriggered: applicationWindow().pageStack.replace(loginPageComponent)
```
Comment (MRZA-MRZA) [inline src/qml/main.qml]: This is active even after user is logged in. I think this should become inactive after login.

Comment (LNJ2) [inline src/qml/main.qml]: I also think so. :D

File: kaidan.qrc (hunk 1 above)
Comment (JBBgameich) [inline kaidan.qrc]: I've done this locally already, will push soon

File: src/qml/main.qml (hunk 5 above)
Comment (LNJ2) [inline src/qml/main.qml]: But then you should first log out and then relogin.

**Comment by LNJ2** [comment]
One thing:
We can't merge this, when not all options you can select in the drawer can be used. E.g. we would want make a release in some days, then we would have a problem... (but the other way, first c++ functions and then the QML-things should be ok)

File: src/qml/main.qml (hunk 5 above)
Comment (JBBgameich) [inline src/qml/main.qml]: Should I remove it? Just wanted to have something in the settings.

**Comment by JBBgameich** [comment]
I've removed everything which  hasn't got any function (locally), it looks a bit poor :D
spectacle ab7840

**Comment by LNJ2** [comment]
@JBBgameich Yeah, that's the problem. I think we should just wait until the most functions are supported by the back-end.

File: src/qml/main.qml [hunk 6]
```diff
@@ -42,12 +60,14 @@ Kirigami.ApplicationWindow {
...
 			// open login page
 			pageStack.pop(); // pop all pages
+			applicationWindow().controlsVisible = false; // disable kirigami controls, for disabling the drawer on Login
 			pageStack.push(loginPage, {"isRetry": true});
 		}
 
 
 		if (kaidan.newLoginNeeded()) {
 			// open login page and get new data from user
+			applicationWindow().controlsVisible = false; // disable kirigami controls, for disabling the drawer on Login
```
Comment (LNJ2) [inline src/qml/main.qml]: Let's keep this uniformly; only use:
`controlsVisible = false;`

... or add the `applicationWindow()....` everywhere.

File: src/qml/main.qml [hunk 7]
```diff
@@ -25,7 +25,25 @@ Kirigami.ApplicationWindow {
 	id: root
 	width: 960
 	height: 540
+	
+	globalDrawer: Kirigami.GlobalDrawer {
```
Comment (LNJ2) [inline src/qml/main.qml]: I think it would be better if we move the drawer into a new file.

**Pull request merged**